### PR TITLE
Fixes #6888 and all other issues related to per_page dropdown

### LINKED ISSUE #6888: Admin: filtering and per-page selections incompatible
Steps to test:

* Visit any admin page that has filtering or quick search and per-page select dropdowns (/admin/orders or /admin/products will work)
* Select a value from the per-page drop down that is not the default and wait for the page to reload
* Enter a value into the quick search or any of the filters on the page, submit the form
* Observe that the per-page selection has been wiped out.
* With the previous search still active, select a per-page value.
* Observe that the per-page value works, but the previous search is wiped out.

Expected outcome:

* You can set a per-page value while searching, and search and your per-page selection is preserved.

Environment:

* Vanilla spree install, Gemfile here: https://gist.github.com/jheilema-nerdery/f189c4508b61e0cc264e
* Can reproduce on Chrome v46; probably also on other browsers

I've fixed this bug in a extension I've been working on, but it requires refactoring the per page and quick search interaction a bit. I know it's possible.

### PR DESCRIPTION
**Pull request by damianlegawiec**
Fixes bugs for admin per page dropdown:
- [X] User selects per page from dropdown -> user uses filtering/searching -> per page rollbacks to default value
- [X] User searches / filters records -> user selects per page from dropdown -> all filters are lost
- [X] feature specs

Related issues: #6888

### CODE REVIEW

**Comment by mbj** [comment]
No specs?

**Comment by damianlegawiec** [comment]
@jitendra @priyank-gupta will add some additional specs as @mbj suggested - there are already ones concerning pagination and so on but some regression ones would also be nice

**Comment by damianlegawiec** [comment]
@jitendra @priyank-gupta specs added

File: backend/app/assets/javascripts/spree/backend/admin.js [hunk 1]
```diff
@@ -139,6 +139,32 @@ jQuery(function($) {
     }
   });
 
+  // per page dropdown
+  // preserves all selected filters / queries supplied by user
+  // changes only per_page value
+  $(".js-per-page-select").change(function() {
+    var form  = $(this).closest(".js-per-page-form");
+    var url   = form.attr('action')
```
Comment (priyank-gupta) [inline backend/app/assets/javascripts/spree/backend/admin.js]: @damianlegawiec line end semicolons missing.

**Comment by priyank-gupta** [comment]
:+1: 
Seems fine to me. Just semicolons at line end needed in js code.

**Pull request merged**